salt/metalk8s/orchestrate/downgrade/post.sls

```diff
@@ -1,4 +1,23 @@
 # Include here all states that should be called after downgrading
 
+{%- set version = pillar.metalk8s.cluster_version %}
+
+{#- The Metalk8s python-kubernetes module for listing objects is bugged for
+    already released versions, The following release versions are affected:
+    2.4.0, 2.4.1, 2.4.2, 2.4.3 and 2.5.0
+    This issue has been fixed here:
+    https://github.com/scality/metalk8s/pull/2593, so we need to skip the
+    PrometheusRule cleanup state for the versions mentioned above. #}
+
+{%- set affected_versions = ['2.4.0', '2.4.1', '2.4.2', '2.4.3', '2.5.0'] %}
+
+{%- if version not in affected_versions %}
 include:
   - metalk8s.addons.prometheus-operator.post-cleanup
+
+{%- else %}
+
+Skipping PrometheusRule cleanup for affected version {{ version }}:
+  test.succeed_without_changes: []
+
+{%- endif %}
```

Comment:
No please, do not do this in the global downgrade post.

Create a file `salt/metalk8s/addons/prometheus-operator/post-downgrade.sls` (and also one for upgrade) and put the logic in the downgrade one

OR

Just put this logic in `post-cleanup` state directly since we will never upgrade with this state and a destination equal to one of the already released one

Reply:
Ok, let's roll with the first approach